src/cluster.c

```diff
@@ -4123,11 +4146,15 @@ sds clusterGenNodeDescription(clusterNode *node) {
     sds ci;
 
     /* Node coordinates */
-    ci = sdscatprintf(sdsempty(),"%.40s %s:%d@%d ",
+    ci = sdscatprintf(sdsempty(),"%.40s %s:%d@%d",
         node->name,
         node->ip,
         node->port,
         node->cport);
+    if (node->hostname && sdslen(node->hostname) > 0) {
```

Comment:
Would be good to add a nodes.conf file with hostname in it and verify it gets loaded fine.